gnovm/pkg/gnolang/preprocess.go

```diff
@@ -308,13 +315,14 @@ func initStaticBlocks(store Store, ctx BlockNode, bn BlockNode) {
 
 				// parent switch statement.
 				ss := ns[len(ns)-1].(*SwitchStmt)
-				// anything declared in ss are copied,
-				// namely ss.VarName if defined.
+				// anything declared in ss.init are copied.
 				for _, n := range ss.GetBlockNames() {
 					last.Predefine(false, n)
 				}
 				if ss.IsTypeSwitch {
 					if ss.VarName != "" {
+						// XXX NameExprTypeDefine in NameExpr?
+						// XXX XXX fix and test.
```

Comment:
this test panics, the name declared in type switch is not correctly captured, as the comment says.

```go
package main

type fooer interface{}

var f fooer = 1

func trymake() func() {
	switch v := f.(type) {
	case int:
		return func() {
			println(v)
		}
	default:
	}
	return func() {}
}
func main() {
	trymake()()
}

// Output:
// 1
```

Reply:
yeah this I will mark as a known issue, something to be fixed later as it is probably uncommon. (see "// XXX NameExprTypeDefine in NameExpr?" in preprocess.go)